odl/tomo/geometry/conebeam.py

```diff
@@ -797,9 +828,30 @@ def __init__(self, apart, dpart, src_radius, det_radius, pitch=0,
         AxisOrientedGeometry.__init__(self, axis)
         # `check_bounds` is needed for both detector and geometry
         check_bounds = kwargs.get('check_bounds', True)
-        detector = Flat2dDetector(dpart, axes=det_axes_init,
-                                  check_bounds=check_bounds)
-        super(ConeFlatGeometry, self).__init__(
+        if det_curvature_radius is None:
+            detector = Flat2dDetector(dpart, axes=det_axes_init,
+                                      check_bounds=check_bounds)
+        else:
+            if np.array(det_curvature_radius).shape == ():
+                detector = CylindricalDetector(dpart,
+                                               radius=det_curvature_radius,
+                                               axes=det_axes_init,
+                                               check_bounds=check_bounds)
+            elif np.array(det_curvature_radius).shape == (2,):
+                if det_curvature_radius[0] != det_curvature_radius[1]:
```

Comment:
Oh now I get it. We don't actually support 2 different radii. In that case, I think the API is a bit awkward, and it would definitely be better to use a single float for spherical and `[r, None]` and `[r, float('inf')]` for cylinders.